source/QuantifierModel/InstantiationPath.cs

```diff
@@ -561,10 +561,6 @@ private void printCycleInfo(InfoPanelContent content, PrettyPrintFormat format)
             if (!hasCycle()) return;
             var cycle = cycleDetector.getCycleQuantifiers();
             var generalizationState = cycleDetector.getGeneralization();
-            if (generalizationState.NeedsIds)

```

Comment:
When is this field set to true in the current code? Is it used for anything else?

Reply:
This field is not used anywhere else.
CycleDetection.NeedsIds is true when:
  - no generalization has been found for the first term nor the last term of the generalized loop
  - the first term and the last term of the generalized loop has no dependent term
  - the loop contains more than one term
 
This can be found in CycleDetection.MarkGeneralizations
(the conditions above are only what I interpreted from the code, it is unclear to me what is the higher-level semantics of NeedsIds)